### ISSUE TITLE
[Substratevm] Provide some doc/sample for using Reflection config files.

### ISSUE THREAD

**Issue by sureshg** [oracle/graal#277]
Right now my application fails with the following error on Mac
```io.airlift.airline.ParseException: Unable to create instance io.airlift.airline.Help
	at java.lang.Throwable.<init>(Throwable.java:287)
	at java.lang.Exception.<init>(Exception.java:84)
	at java.lang.RuntimeException.<init>(RuntimeException.java:80)
	at io.airlift.airline.ParseException.<init>(ParseException.java:31)
	at io.airlift.airline.ParserUtil.createInstance(ParserUtil.java:22)
	at io.airlift.airline.DefaultCommandFactory.createInstance(DefaultCommandFactory.java:10)
	at io.airlift.airline.ParserUtil.createInstance(ParserUtil.java:38)
	at io.airlift.airline.Cli.parse(Cli.java:136)
	at io.airlift.airline.Cli.parse(Cli.java:108)
	at io.airlift.airline.Cli.parse(Cli.java:103)
	at com.oneops.secrets.Main.main(Main.java:84)
	at com.oracle.svm.core.JavaMainWrapper.run(JavaMainWrapper.java:155)
	at Lcom/oracle/svm/core/code/CEntryPointCallStubs;.com_002eoracle_002esvm_002ecore_002eJavaMainWrapper_002erun_0028int_002corg_002egraalvm_002enativeimage_002ec_002etype_002eCCharPointerPointer_0029(generated:0)
Caused by: java.lang.NoSuchMethodException: io.airlift.airline.Help.<init>()
```

Further exploration revealed that code internally uses reflection to create new instance

```public static <T> T createInstance(Class<T> type)
    {
        if (type != null) {
            try {
                return type.getConstructor().newInstance();
            }
            catch (Exception e) {
                throw new ParseException(e, "Unable to create instance %s", type.getName());
            }
        }
        return null;
    }
```

So it would be great if some documentation and a sample `ReflectionConfigurationFiles` made available for reference.

**Comment by peter-hofer** [comment]
There is a sample file for `ReflectionConfigurationFiles` that is shown by `-H:+PrintFlags`. Does that help you?
```
Files containing JSON arrays describing program elements to be made available for reflection.
The JSON object schema is:

    Class {
      String name; // fully qualified class name
      boolean allDeclaredMethods; // Class.getDeclaredMethods() will be available
      boolean allPublicMethods; // Class.getMethods() will be available
      boolean allDeclaredFields; // Class.getDeclaredFields() will be available
      boolean allPublicFields; // Class.getFields() will be available
      {   
        String name; // method name
        String[] parameterTypes; // method sig (only required if name is ambiguous)
      }[] methods;
      {   
        String name; // field name
      }[] fields;
    }

For example:

	[
	  {
	    "name" : "java.lang.Class",
	    "allDeclaredMethods" : "true",
	    "allPublicMethods" : "true"
	  },
	  {
	    "name" : "java.util.HashMap",
	    "fields" : [
	      { "name" : "loadFactor" },
	      { "name" : "threshold" }
	    ],
	    "methods" : [
	      { "name" : "keySet" },
	      { "name" : "<init>", "parameterTypes" : ["int"] },
	    ]
	  }
	]

```

**Comment by peter-hofer** [comment]
Sure. I'll also see if I can extend the sample with more diverse method signatures (`parameterTypes`).

**Comment by peter-hofer** [comment]
Resolved in 85b38407ae31f93f4817a241d28921f8a79c4556. We also support a nicer syntax for array parameter types now, like `int[]` or `java.lang.String[][]` (before: `[I`, `[[Ljava/lang/String;`)
